sdccc/src/main/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ConditionalPreconditions.java

```diff
@@ -688,6 +726,38 @@ static boolean manipulation(final Injector injector) {
         }
     }
 
+    /**
+     * Precondition which triggers all possible description changes with parent-child relationships.
+     */
+    public static class DescriptionModificationAllWithParentChildRelationshipPrecondition extends SimplePrecondition {
+
+        private static final Logger LOG = LogManager.getLogger(DescriptionModificationAllWithParentChildRelationshipPrecondition.class);
+
+        /**
+         * Creates a description changed precondition check.
+         */
+        public DescriptionModificationAllWithParentChildRelationshipPrecondition() {
+            super(DescriptionModificationAllWithParentChildRelationshipPrecondition::preconditionCheck, DescriptionModificationAllWithParentChildRelationshipPrecondition::manipulation);
+        }
+
+        static boolean preconditionCheck(final Injector injector) {
+            return false; // always trigger description changes
+        }
+
+        /**
+         * Performs the removal, reinsertion and update of all modifiable descriptors in the mdib to trigger reports.
+         *
+         * @param injector to analyze mdib on
+         * @return true if successful, false otherwise
+         * @throws PreconditionException on errors
+         */
+        static boolean manipulation(final Injector injector) {
+            boolean result1 = descriptionModificationManipulation(injector, LOG);
+            boolean result2 = descriptionUpdateWithParentChildRelationshipManipulation(injector, LOG);
+            return result1 && result2;
```

Comment:
Doesn't this require the device to still support getRemovableDescriptors? `descriptionModificationManipulation` will always return false, when no removable descriptors are present, or no successful insert / delete manipulation was seen.

Reply:
It becomes relevant at the latest when you add unit tests. You should add unit tests for the cases:
- `descriptionModificationManipulation` is satisfied but `descriptionUpdateWithParentChildRelationshipManipulation` is not
- `descriptionUpdateWithParentChildRelationshipManipulation` is satisfied but `descriptionModificationManipulation` is not
- both are not satisfied